### PR TITLE
Complete typescript

### PR DESCRIPTION
**Pull request by zone117x**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,2 @@
-jest.config.js
\ No newline at end of file
+jest.config.js
+converage/*
```
Comment (kyranjamie) [inline .eslintignore]: is this a typo?

File: .eslintrc.js [hunk 2]
```diff
@@ -24,20 +25,20 @@ module.exports = {
...
     "import/prefer-default-export": "off",
     "max-len": [1, { code: 120 }],
-    "@typescript-eslint/no-var-requires": [1],
+    "@typescript-eslint/no-var-requires": "error",
     '@typescript-eslint/member-delimiter-style': [
       'error',
       {
         multiline: {
-          delimiter: 'none',
+          delimiter: 'semi',
```
Comment (kyranjamie) [inline .eslintrc.js]: Would be great to use @blockstack/prettier-config too ✌🏼

File: .eslintignore (hunk 1 above)
Comment (zone117x) [reply .eslintignore]: Yes, thank!

**Comment by jeffdomke** [comment]
@hstove can you please take a look

File: lib/aggregators/blocks.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ class BlocksAggregator extends Aggregator {
         { total: blocks.length }
       );
     }
-    return Promise.map(blocks, async (_block) => {
+    return BluebirdPromise.map(blocks, async (_block: any) => {
```
Comment (yknl) [inline lib/aggregators/blocks.ts]: What's the advantage of using BlueBirdPromise over native promises here?

File: lib/bitcore-db/queries.ts [hunk 4]
```diff
@@ -51,6 +51,7 @@ export const getBlocks = async (date: string, page = 0): Promise<Block[]> => {
 
   const blocks: Block[] = blocksResult.map(block => ({
     ...block,
+    time: block.time.getTime() / 1000,
```
Comment (yknl) [inline lib/bitcore-db/queries.ts]: Is this being used some where right now or just adding for future use?

File: lib/aggregators/blocks.ts (hunk 3 above)
Comment (zone117x) [reply lib/aggregators/blocks.ts]: Native Promise does not have a `map` function -- the `Promise` global variable was previously overridden / decorated.

File: lib/bitcore-db/queries.ts (hunk 4 above)
Comment (zone117x) [reply lib/bitcore-db/queries.ts]: The other `Block` queries specify `time`. This change makes the `Block` type consistent so callers never have to worry about it being undefined.

**Pull request merged**